Minor tweak in extract_bibtex()

RLumSK opened the pull request:
Fix wrongly formatted BibTeX strings that could occur under particular circumstances. 

## Description

This PR addresses two issues I became aware of while using a script to extract ORCID records that 
was working flawlessly before but crashed out of a sudden: 

1. In the original version of the parser in `extract_bibtext()` I used `nchar()` to count the number of characters in the already extracted BibTeX string to correct the final line of the BibTeX entry. The default partial matching mode is `nchar(...,type = "chars")`. I believe that due to recent changes in the UTF-8 character handling in R, the number `nchar()` returns changed for some cases (as it is not the some if `nchar(..., type = "bytes")`. In consequence, instead of having a nice ending of a BibTeX record, the record was even wrongly formated with a double `}}`.  Regardless the cause, my handling of this matter was wrong in the first place, because what I wanted to do (but did not) was `nchar(..., type = "width")`. I corrected the code accordingly. 
2. While running the related test, it turned out that the put code I had used does not exist anymore, I added a working put code. 

I tested the fix with R-devel and R-release.

## Related Issue
Nothing I am aware of.

## Example

The current version:

```r
x <- strsplit(x = orcid_auth(), " ")[[1]][2]
Sys.setenv(ORCID_TOKEN = x)
aa <- rorcid::orcid_citations(orcid = "0000-0002-0734-2199", put_code = "77226586")
aa$citation
```

The last characters show

```
,}\n}" 
```

while it should read (the new behaviour after the fix).

```
,\n}" 
```

RLumSK committed:
Fix last line problem and fix test, which did not work anymore because the put code was invalid

RLumSK committed:
Add NEWS

RLumSK committed:
Re-recording the cassette

RLumSK commented:
@sckott Sorry for being an idiot, but I guess I don't understand why the GitHub Actions do not pass here. After re-recording the cassette it works just fine my computer with the tests, but here something seems to be missing regarding the ORCID authentication. Would you mind helping me out here?

sckott commented:
thanks @RLumSK !

wrt checks not passing, secrets are not available on pull requests - see e.g., https://github.com/ropensci/rorcid/pull/92/checks?check_run_id=2643315766#step:10:106

i'll have a look

RLumSK commented:
> wrt checks not passing, secrets are not available on pull requests - see, e.g., https://github.com/ropensci/rorcid/pull/92/checks?check_run_id=2643315766#step:10:106

@sckott Thanks, I saw that but I guess I don't understand why the secret is not available for the pull request. Isn't it something you would store directly as a GitHub secret? I am just curious so that I can make it correctly the next time.

sckott commented:
secrets are stored here on github, correct. however, it is a github feature that secrets are not available in pull requests for security reasons, see the docs 

https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

> In order to protect public repositories for malicious users we run all pull request workflows raised from repository forks with a read-only token and no access to secrets.

If your changes in the PR pass locally for you thats a good check - i always check PRs locally as well, so that's another check.

sckott commented on NEWS.md:
```diff
@@ -3,8 +3,9 @@ rorcid 0.7.0
-* `orcid_search()` changes: now returns number of records found as an attribute; get it by doing `attr(x, "found")` if `x` is the result of a call to `orcid_search()`. In addition, added an example of retreiving number of records found, as well as added more documentation to the function page on pagination. (#86)
+* `orcid_search()` changes: now returns number of records found as an attribute; get it by doing `attr(x, "found")` if `x` is the result of a call to `orcid_search()`. In addition, added an example of retrieving number of records found, as well as added more documentation to the function page on pagination. (#86)
 * Auth gains a 3rd method: 2 legged OAuth. Updated `?orcid_auth` docs on the 3 different auth methods, as well as the README and auth vignette.  (#87)
+* `orcid_citations()` produced wrongly formatted BibTeX strings under particular circumstances; fixed. 
```
add the PR in parens at the end of the line, like `(#92)`

RLumSK replied:
Done

sckott reviewed (changes requested):
thanks, just a few comments. 

Also, could you add a test specifically for the change you made if possible for `orcid_citations `?

RLumSK committed:
+ Amend NEWS
+ Add better test to make sure such things do not remain undetected in the future

RLumSK commented:
> secrets are stored here on github, correct. however, it is a github feature that secrets are not available in pull requests for security reasons, see the docs

Thanks, it makes sense.

sckott commented:
thanks!
